File: 448-FindAllNumbersDisappearedInAnArray/448-FindAllNumbersDisappearedInAnArray.js
// Last updated: 4/29/2026, 10:30:14 AM
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDisappearedNumbers = function(nums) {
    for(let i = 0; i < nums.length; i++) {
        const num = Math.abs(nums[i]);
        const idx = num - 1;
        if(nums[idx] > 0) {
            nums[idx] = -nums[idx];
        }
    }
    const res = [];
    for(let i = 0; i < nums.length; i++) {
        if(nums[i] > 0) {
            res.push(i+1);
        }
    }
    return res;
};

File: 448-FindAllNumbersDisappearedInAnArray/448-FindAllNumbersDisappearedInAnArray.py
# Last updated: 4/29/2026, 10:30:08 AM
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ref = [0] * n
        for num in nums:
            ref[num - 1] = 1
        print(ref)
        ret = []
        for idx, item in enumerate(ref):
            print(f"idx: {idx}, item: {item}")
            if item == 0:
                ret.append(idx + 1)
        
        return ret